### PR TITLE
Add the models

### PR DESCRIPTION
**Pull request by advaypal**
Still WIP. DO NOT MERGE

### CODE REVIEW

**Comment by advaypal** [comment]
Guys, this is a lot of the code for the models and the data storage, tell me what you guys think and whether I should make changes?
@anand-sundaram @Tzyinc Let me know when you have the serialisation function and audio struct ready respectively, will update accordingly.

**Comment by anand-sundaram** [comment]
@advaypal I have added the serialization function in #5 
You can try rebasing this branch on top of that if it looks okay to you.

File: MIDIchlorians/MIDIchlorians/Pad.swift
Comment (ngzhian) [inline MIDIchlorians/MIDIchlorians/Pad.swift:31]: what's this comment about?

File: MIDIchlorians/MIDIchlorians/Pad.swift
Comment (ngzhian) [inline MIDIchlorians/MIDIchlorians/Pad.swift]: does this need to be implemented?

File: MIDIchlorians/MIDIchlorians/Session.swift
Comment (ngzhian) [inline MIDIchlorians/MIDIchlorians/Session.swift]: let's put it at 128 for now, we might not even get to implement this

File: MIDIchlorians/MIDIchlorians/Session.swift
Comment (ngzhian) [inline MIDIchlorians/MIDIchlorians/Session.swift]: use constants from config? same as fields below `numRows`, `numCols`

File: MIDIchlorians/MIDIchlorians/Session.swift
Comment (ngzhian) [reply MIDIchlorians/MIDIchlorians/Session.swift]: also, put this constant in config

File: MIDIchlorians/MIDIchlorians/Session.swift
Comment (ngzhian) [inline MIDIchlorians/MIDIchlorians/Session.swift:102]: think we need to reset `padList` everytime we call this `prepareForSave`, otherwise we keep appending pads.

File: MIDIchlorians/MIDIchlorians/Session.swift
Comment (ngzhian) [inline MIDIchlorians/MIDIchlorians/Session.swift]: reset pads to `[]`?

File: MIDIchlorians/MIDIchlorians/SessionSaver.swift
Comment (ngzhian) [inline MIDIchlorians/MIDIchlorians/SessionSaver.swift]: extract this string into constant
and use `.first`

**Review by ngzhian — CHANGES_REQUESTED** [review]
Looks great, check out inline comments related to constants and `prepareForSave`.

File: MIDIchlorians/MIDIchlorians/Pad.swift
Comment (advaypal) [reply MIDIchlorians/MIDIchlorians/Pad.swift]: Yup, will call the serialisation functions here

File: MIDIchlorians/MIDIchlorians/Session.swift
Comment (advaypal) [reply MIDIchlorians/MIDIchlorians/Session.swift]: Got it

File: MIDIchlorians/MIDIchlorians/Session.swift
Comment (advaypal) [reply MIDIchlorians/MIDIchlorians/Session.swift:102]: (y)

File: MIDIchlorians/MIDIchlorians/Session.swift
Comment (advaypal) [reply MIDIchlorians/MIDIchlorians/Session.swift]: Ok

File: MIDIchlorians/MIDIchlorians/SessionSaver.swift
Comment (advaypal) [reply MIDIchlorians/MIDIchlorians/SessionSaver.swift]: Ok

File: MIDIchlorians/MIDIchlorians/Pad.swift
Comment (advaypal) [reply MIDIchlorians/MIDIchlorians/Pad.swift:31]: Need to add the name of the audio property here, to tell Realm that it doesn't need to be saved

Comment (advaypal) [reply MIDIchlorians/MIDIchlorians/Pad.swift:31]: Doesn't matter if we're only storing audio files

**Comment by advaypal** [comment]
@ngzhian Take a look at the changes. Have added the saving/loading methods which currently do not work, but can be merged in so that everyone can work with them

**Comment by advaypal** [comment]
Don't merge it in yet though, I'll merge it once you've reviewed. Had temporarily removed SwiftLint and am adding it back now

**Pull request merged**
